File: languages/ruby/src/provider/LanguageSortAnalysis.rb
require './src/model/TableAnalysis.rb'
require './src/model/UserInfo.rb'
require './src/model/exception/InvalidParameterException.rb'

# Classe para analise de dados, obtem um usuario aleatoriamente
# @see TableAnalysis
class LanguageSortAnalysis
    include TableAnalysis

    # Realiza uma analise aleatoria dos dados
    # @param userInfoList Lista de dados a ser analisada
    # @return Elemento aleatorio da lista
    # @see TableAnalysis
    def analysis(userInfoList)
        if(userInfoList.nil? || userInfoList.size == 0)
            raise InvalidParameterException.new("'userInfoList' é nil ou vazio")
        end

        sortedList = userInfoList.sort_by do |u1| 
            u1.getCredit
        end

        return sortedList
    end
end

File: languages/python3/src/provider/LanguageSortAnalysis.py
from src.model.TableAnalysis import TableAnalysis
from src.model.UserInfo import UserInfo
from src.model.exception.InvalidParameterException import InvalidParameterException

# Classe para analise de dados, obtem um usuario aleatoriamente
# @see TableAnalysis
class LanguageSortAnalysis(TableAnalysis):

    # Realiza uma analise aleatoria dos dados
    # @param userInfoList Lista de dados a ser analisada
    # @return Elemento aleatorio da lista
    # @see TableAnalysis
    def analysis(self, userInfoList):
        if(userInfoList == None or len(userInfoList) == 0):
            raise InvalidParameterException("'userInfoList' é None ou vazio")

        sortedList = sorted(userInfoList, key = lambda userInfo : userInfo.getCredit())
        return sortedList